src/convertToObject.js

```diff
@@ -5,8 +5,30 @@
  *
  * @return {object}
  */
+
 function convertToObject(sourceString) {
-  // write your code here
+  const delEmptyLine = sourceString.replace(/\n/g, '');
```

Comment:
[CODE KNOWLEDGE] - Don't need to make copy before each operation. Moreover, methods like `filter` or `map` don't mutate original array. Instead of replacing all newline characters in `sourceString`, you can directly split the string into lines and trim those. This will also eliminate the need for the `delEmptyLine` variable.